src/Browscap/Filter/CustomFilter.php

```diff
@@ -35,26 +35,21 @@ class CustomFilter implements FilterInterface
     private $propertyHolder;
 
     /**
-     * @param array                              $fields
-     * @param \Browscap\Data\PropertyHolder|null $propertyHolder
+     * @param \Browscap\Data\PropertyHolder $propertyHolder
+     * @param array                         $fields
      */
-    public function __construct(array $fields, PropertyHolder $propertyHolder = null)
+    public function __construct(PropertyHolder $propertyHolder, array $fields)
     {
-        $this->fields = $fields;
-
-        if (null === $propertyHolder) {
-            $this->propertyHolder = new PropertyHolder();
-        } else {
-            $this->propertyHolder = $propertyHolder;
-        }
+        $this->fields         = $fields;
+        $this->propertyHolder = $propertyHolder;
     }
 
     /**
      * returns the Type of the filter
      *
      * @return string
      */
-    public function getType()
+    public function getType() : string
     {
         return 'CUSTOM';
```

Comment:
Perhaps another case where a constant should be used?

Reply:
+1 for constant